javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -709,14 +709,14 @@ public int indexOf(T element, int from) {
 
     @Override
     public Vector<T> insertAll(int index, Iterable<? extends T> elements) {
-        if ((index < 0) || (index > length())) {
-            throw new IndexOutOfBoundsException("insert(" + index + ", e) on Vector of length " + length());
-        } else {
```

Comment:
Generally we could probably avoid some jumps if we put the more likely cases first, otherwise we could have a jump to the else statement and a return separately.
But than again, sometimes this is done automatically by JIT.
However, humans work in that way also, i.e. they expect the more frequent cases to appear first.

If it bothers you, I can revert these.